src/test/codegen/export-no-mangle.rs

```diff
@@ -10,6 +10,9 @@
 
 #![crate_type = "lib"]
 
+// Avoid functions being merged through calls to this dummy
```

Comment:
This test seems to be a good candidate for `-C no-prepopulate-passes`, since AFAIK it tests rustc's IR generation rather than LLVM optimizations. With that flag added, this hack shouldn't be necessary.